File: app/src/main/java/com/umbrella/budgetapp/Customs/DetailedProgressbar.java
package com.umbrella.budgetapp.Customs;

import android.content.Context;
import android.content.res.ColorStateList;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.umbrella.budgetapp.R;

public class DetailedProgressbar extends RelativeLayout {

    private TextView titleText;
    private TextView valueText;
    private ProgressBar progressBar;

    public DetailedProgressbar(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }
    
    private void init(Context context, AttributeSet attrs) {
        inflate(context, R.layout.custom_detailed_progressbar, this);

        TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.DetailedProgressbar);

        CharSequence title = typedArray.getText(R.styleable.DetailedProgressbar_title);
        CharSequence value = typedArray.getText(R.styleable.DetailedProgressbar_value);
        int progress = typedArray.getInteger(R.styleable.DetailedProgressbar_progress_value, 0);
        int colors = typedArray.getColor(R.styleable.DetailedProgressbar_progress_color, getResources().getColor(R.color.colorAccent, null));

        typedArray.recycle();

        titleText = findViewById(R.id.Progressbar_Name);
        valueText = findViewById(R.id.Progressbar_Amount);
        progressBar = findViewById(R.id.Progressbar_Attr);

        setTitleText(title);
        setValueText(value);
        setProgressBar(progress, colors);
    }

    public CharSequence getTitleText() {
        return titleText.getText();
    }
    public CharSequence getValueText() {
        return valueText.getText();
    }
    public int getProgressBar() {
        return progressBar.getProgress();
    }
    public ColorStateList getProgressColor() {
        return  progressBar.getProgressTintList();
    }

    public void setTitleText(CharSequence value) {
        titleText.setText(value);
    }
    public void setValueText(CharSequence value) {
        valueText.setText(value);
    }
    public void setProgressBar(int value, int coloring) {
        progressBar.setProgress(value);
        progressBar.setProgressTintList(ColorStateList.valueOf(coloring));
    }

}

File: app/src/main/java/com/umbrella/budgetapp/ui/components/DetailedProgressbar.kt
package com.umbrella.budgetapp.ui.components

import android.content.Context
import android.content.res.ColorStateList
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.ProgressBar
import android.widget.RelativeLayout
import android.widget.TextView
import com.umbrella.budgetapp.R
import com.umbrella.budgetapp.cache.Memory
import com.umbrella.budgetapp.extensions.currencyText
import java.math.BigDecimal

class DetailedProgressbar @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : RelativeLayout(context, attrs, defStyleAttr) {
    private var titleView: TextView
    private var valueView: TextView
    private var progressbar: ProgressBar

    var title: String? = ""
    set(value) {
        field = value
        titleView.text = value
    }

    var value : BigDecimal? = BigDecimal.ZERO
    set(value) {
        field = value
        valueView.currencyText(Memory.lastUsedCountry.symbol, value ?: BigDecimal.ZERO)
    }

    var progress : Int = 0
    set(value) {
        field = value
        progressbar.progress = value
    }

    var max : Int = 100
    set(value) {
        field = value
        progressbar.max = value
    }

    var progressBarColor : Int = resources.getColor(R.color.colorAccent, context.theme)
    set(value) {
        field = value
        progressbar.progressTintList = ColorStateList.valueOf(value)
    }

    init {
        LayoutInflater.from(context).inflate(R.layout.custom_detailed_progressbar, this, true)

        titleView = findViewById(R.id.Progressbar_Name)
        valueView = findViewById(R.id.Progressbar_Amount)
        progressbar = findViewById(R.id.Progressbar_Attr)

        attrs?.let {
            val styledAttributes = context.obtainStyledAttributes(it, R.styleable.DetailedProgressbar, 0, 0)

            title = styledAttributes.getString(R.styleable.DetailedProgressbar_title)
            value = styledAttributes.getString(R.styleable.DetailedProgressbar_value)?.toBigDecimal()
            progress = styledAttributes.getInteger(R.styleable.DetailedProgressbar_progress_value, 0)
            progressBarColor = styledAttributes.getColor(R.styleable.DetailedProgressbar_progress_value, resources.getColor(R.color.colorAccent, context.theme))

            styledAttributes.recycle()
        }
    }
}